avocado/plugins/diff.py

```diff
@@ -329,17 +329,10 @@ def _get_job_data(jobdir):
 
     @staticmethod
     def _setup_job(job_id):
-        if os.path.isdir(job_id):
-            resultsdir = os.path.expanduser(job_id)
-            job_id = ''
-        elif os.path.isfile(job_id):
-            resultsdir = os.path.dirname(os.path.expanduser(job_id))
-            job_id = ''
-        else:
-            logs_dir = settings.get_value('datadir.paths', 'logs_dir',
-                                          default=None)
-            logdir = os.path.expanduser(logs_dir)
-            resultsdir = jobdata.get_resultsdir(logdir, job_id)
+        logs_dir = settings.get_value('datadir.paths', 'logs_dir',
+                                      default=None)
+        logdir = os.path.expanduser(logs_dir)
```

Comment:
This is actually not needed, you use `expanduser` later in the get_resultsdir`. The `expanduser` is quite useful when dealing with some libraries, but you don't need to use it everywhere...